### PR TITLE
Updated dining tests 

### LINKED ISSUE #164: Broken/outdated tests for dining module
@akrishan11 @nij-patel The PR #161 breaks the dining module tests. See the errors that I get in #163:
```
FAILED tests/dining_test.py::DiningTest::test_get_dining_locations - requests.exceptions.ConnectionError: Connection refused by Responses - the call doesn't match any registered mock.

Request: 
- GET https://api.dineoncampus.com/v1/locations/status?site_id=5e6fcc641ca48e0cacd93b04&platform=

Available matches:
- GET https://m.pitt.edu/dining/index.json?feed=dining_locations&_kgoui_object=kgoui_Rcontent_I2&start=0 URL does not match
- GET https://m.pitt.edu/dining/index.json?feed=dining_locations&_kgoui_object=kgoui_Rcontent_I2&start=10 URL does not match
- GET https://m.pitt.edu/dining/index.json?feed=dining_locations&_kgoui_object=kgoui_Rcontent_I2&start=20 URL does not match
FAILED tests/dining_test.py::DiningTest::test_get_dining_locations_by_status_closed - AttributeError: module 'pittapi.dining' has no attribute 'get_locations_by_status'
FAILED tests/dining_test.py::DiningTest::test_get_dining_locations_by_status_open - AttributeError: module 'pittapi.dining' has no attribute 'get_locations_by_status'
```
For instance, the PR deleted the `get_dining_locations_by_status` function, so that automatically breaks two of the three tests in `dining_test.py`. I think the first test is failing because of changed URLs that weren't likewise updated in the test.

In any case, the unit tests need to be updated to properly test the new API, and all new PRs will be blocked until then. We can either put all PRs on hold until these new tests are written, or we can temporarily disable testing for the dining module for the time being.

### PR DESCRIPTION
**Pull request by akrishan11**
Fixes #164

Updated tests for functions in new Dining module

Added mocks for dining locations, dining menu, and dining schedule

### CODE REVIEW

**Comment by tianyizheng02** [comment]
Edited the PR description to automatically [link the PR to the correct issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)

File: tests/dining_test.py [hunk 1]
```diff
@@ -31,50 +32,42 @@
...
-                      'https://m.pitt.edu/dining/index.json?feed=dining_locations&_kgoui_object=kgoui_Rcontent_I2&start=20',
-                      json=self.dining_data, status=200)
-        locations = dining.get_locations_by_status('open')
-        self.assertIsInstance(locations, list)
-        for location in locations:
-            self.assertEqual(location['status'], 'open')
+                      "https://api.dineoncampus.com/v1/locations/weekly_schedule?site_id=5e6fcc641ca48e0cacd93b04&date=%222024-04-12%22",
+                      json=self.dining_schedule_data, status=200)
+
+        self.assertIsInstance(dining.get_location_hours("The Eatery", datetime.datetime(2024,4,12)), dict)
```
Comment (tianyizheng02) [inline tests/dining_test.py]:
```suggestion
        self.assertIsInstance(dining.get_location_hours("The Eatery", datetime.datetime(2024, 4, 12)), dict)
```
Minor PEP8 fixes

File: tests/dining_test.py [hunk 2]
```diff
@@ -31,50 +32,42 @@
...
         responses.add(responses.GET,
-                      'https://m.pitt.edu/dining/index.json?feed=dining_locations&_kgoui_object=kgoui_Rcontent_I2&start=20',
-                      json=self.dining_data, status=200)
-        locations = dining.get_locations_by_status('closed')
-        self.assertIsInstance(locations, list)
-        for location in locations:
-            self.assertEqual(location['status'], 'closed')
+                      "https://api.dineoncampus.com/v1/location/610b1f78e82971147c9f8ba5/periods/659daa4d351d53068df67835?platform=0&date=24-04-12",
+                      json=self.dining_menu_data, status=200)
+        locations = dining.get_location_menu("The Eatery", datetime.datetime(2024,4,12), "Breakfast")
```
Comment (tianyizheng02) [inline tests/dining_test.py]:
```suggestion
        locations = dining.get_location_menu("The Eatery", datetime.datetime(2024, 4, 12), "Breakfast")
```
Minor PEP8 fixes

**Review by tianyizheng02 — COMMENTED** [review]
Tests look good, I think. Where do the contents of the new JSON files come from, BTW?

**Comment by tianyizheng02** [comment]
```
=========================== short test summary info ============================
FAILED tests/lab_test.py::LabTest::test_fetch_labs - RecursionError: maximum recursion depth exceeded
FAILED tests/lab_test.py::LabTest::test_get_status - RecursionError: maximum recursion depth exceeded
FAILED tests/library_test.py::LibraryTest::test_get_documents - RecursionError: maximum recursion depth exceeded
FAILED tests/people_test.py::PeopleTest::test_people_get_person - RecursionError: maximum recursion depth exceeded
FAILED tests/people_test.py::PeopleTest::test_people_get_person_none - RecursionError: maximum recursion depth exceeded
FAILED tests/people_test.py::PeopleTest::test_people_get_person_too_many - RecursionError: maximum recursion depth exceeded
=================== 6 failed, 44 passed, 2 skipped in 4.28s ====================
```
The tests that are failing don't have anything to do with the dining module, gonna try to take a look at the code to see if there are any patterns.

**Comment by akrishan11** [comment]
> Tests look good, I think. Where do the contents of the new JSON files come from, BTW?

I just copied the exact data that would be returned from the Dine on Campus at the specific endpoints called by the sample function calls. I don't have much experience writing tests so I'm not sure if there was anything wrong with my approach.

**Comment by tianyizheng02** [comment]
> The tests that are failing don't have anything to do with the dining module, gonna try to take a look at the code to see if there are any patterns.

OK, I'm pretty sure the failing tests are failing because their inputs aren't mocked. The failing tests in `lab_test.py`, `library_test.py`, and `people_test.py` all run the module functions directly (i.e. make HTTP requests) as opposed to calling the function with mocked inputs. All function tests for other modules (that aren't skipped) _do_ mock their inputs, and those are clearly passing. My guess is that real HTTP requests get messed up when the enclosing functions are run from a test environment, probably something to do with shared resources in the test environment.

TL;DR the tests in `lab_test.py`, `library_test.py`, and `people_test.py` all should be rewritten to use mocked inputs instead of making real HTTP requests.

**Review by nij-patel — APPROVED** [review]
